samples/hazelcast/src/main/java/sample/Config.java

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+package sample;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+import javax.servlet.ServletContext;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.session.MapSessionRepository;
+import org.springframework.session.SessionRepository;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.web.http.SessionRepositoryFilter;
+
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.config.SerializerConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+
+// tag::class[]
+@Configuration
+public class Config {
+
+ 	private String sessionMapName = "spring:session:sessions";
+ 	
+	@Autowired
+	private ApplicationEventPublisher eventPublisher;
+	
+ 	@Bean(destroyMethod = "shutdown")
+ 	public HazelcastInstance hazelcastInstance() {
+ 		com.hazelcast.config.Config cfg = new com.hazelcast.config.Config();
+		NetworkConfig netConfig = new NetworkConfig();
+		netConfig.setPort(getAvailablePort());
+		cfg.setNetworkConfig(netConfig);
+		SerializerConfig serializer = new SerializerConfig()
+			.setTypeClass(Object.class)
+			.setImplementation(new ObjectStreamSerializer());
+		cfg.getSerializationConfig().addSerializerConfig(serializer);
+		MapConfig mc = new MapConfig();
+		mc.setName(sessionMapName);
+
+		mc.setMaxIdleSeconds(60);
+		cfg.addMapConfig(mc);
+
+		return Hazelcast.newHazelcastInstance(cfg);
+ 	}
+ 	
+ 	@Bean
+ 	public SessionRemovedListener removeListener() {
+ 		return new SessionRemovedListener(eventPublisher);
+ 	}
+ 	
+ 	@Bean
+ 	public SessionEvictedListener evictListener() {
+ 		return new SessionEvictedListener(eventPublisher);
+ 	}
+ 	
+ 	@Bean
+ 	public SessionCreatedListener addListener() {
+ 		return new SessionCreatedListener(eventPublisher);
+ 	}
+ 	
+ 	@Bean
+	public MapSessionRepository sessionRepository(HazelcastInstance instance,
+			                                      SessionRemovedListener removeListener,
+			                                      SessionEvictedListener evictListener,
+			                                      SessionCreatedListener addListener) {
+ 		IMap<String,ExpiringSession> sessions = instance.getMap(sessionMapName);
+ 		sessions.addEntryListener(removeListener, true);
+ 		sessions.addEntryListener(evictListener, true);
+ 		sessions.addEntryListener(addListener, true);
+		return new MapSessionRepository(sessions);
+	}
+ 	
+	@Bean
+	public <S extends ExpiringSession> SessionRepositoryFilter<? extends ExpiringSession> springSessionRepositoryFilter(SessionRepository<S> sessionRepository, ServletContext servletContext) {
```

Comment:
Master has a newly added `@EnableSpringHttpSession` which will take care of the `SessionRepositoryFilter` and adding a `SessionEventHttpSessionListenerAdapter`. It also allows defaulting and overriding `HttpSessionStrategy`.